### DISCUSSION TITLE
How does the 16-bit parallel mode work?

### DISCUSSION THREAD

**Discussion by pastaskrue** [lovyan03/LovyanGFX#351] [General]
Hello there, I know that this is a huge question and maybe there wont be any answers, but, how does the 16-bit parallel mode work?
I have been looking at this library and some others to get an understanding of how this comes together but I cannot seem to figure out how.
When you select to choose the 16-bit parallel mode, which files then activates (broad question I know, sorry!) and what does it do different from the 8-bit mode. If anyone can draw a path of how the code moves through the files it would be great! And is there a difference between the different LCD drivers of how to implement parallel mode?

Example:
User_Setup.h -> Panel_ILI948x.hpp -> ?

I know it is a lot to ask, sorry!

Best regards

**Comment by lovyan03** [comment]
Hello, @Googool 
Currently Bus_Parallel16 is implemented only in ESP32 S2 and S3.
I would like to integrate Bus_Parallel8 and Bus_Parallel16 if possible, but have given up as it would be a bit too much of a design change to do so with the current version.

The difference between 8-bit and 16-bit has a particularly large impact on the processing of odd lengths when operating in 24-bit color mode.

e.g. use writePixels function...
 - In 16-bit color mode, the data length is always even, regardless of whether the number of pixels is even or odd.
 - In 24-bit color mode, if the number of pixels is odd, the data length is also odd.

Bus_Parallel8 has no particular effect, but Bus_Parallel16 holds the last 1Byte without sending if the transmission length is odd.
If writePixels are subsequently reused, a set of pending 1Bytes is sent at the beginning of the new data.
If a command is requested after writePixels, it sends the requested command after sending the pending 1Byte and 0x00 set.

This is because if the last 1Byte is sent without pending, there will be a data gap when writePixels are reused.
